tests/backup/backup_nfs_test.go

```diff
@@ -203,3 +205,222 @@ var _ = Describe("{DeleteNfsExecutorPodWhileBackupAndRestoreInProgress}", func()
+				log.FailOnError(err, fmt.Sprintf("Getting UID for backup %v", backupName))
+				backupInspectRequest := &api.BackupInspectRequest{
+					Name:  backupName,
+					Uid:   backupUID,
+					OrgId: orgID,
+				}
+				resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+				log.FailOnError(err, fmt.Sprintf("error inspecting backup %v", backupName))
+				currentBackupPath := globalBucket + "/" + resp.Backup.BackupPath
+				log.Infof("Deleting the JSON files from the NFS backup location for backup %v", backupName)
+				DeleteJSONFilesFromNFSLocation(currentBackupPath)
+			}
+		})
+
+		Step("Verify if the backups are in CloudBackupMissing state after the JSON file deletion", func() {
+			log.InfoD("Verify if the backups are in CloudBackupMissing state after the JSON file deletion")
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, backupName := range backupNames {
+				wg.Add(1)
+				go func(backupName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					bkpUid, err := Inst().Backup.GetBackupUID(ctx, backupName, orgID)
+					log.FailOnError(err, "Fetching backup uid")
+					backupInspectRequest := &api.BackupInspectRequest{
+						Name:  backupName,
+						Uid:   bkpUid,
+						OrgId: orgID,
+					}
+					requiredStatus := api.BackupInfo_StatusInfo_CloudBackupMissing
+					backupCloudBackupMissingCheckFunc := func() (interface{}, bool, error) {
+						resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+						if err != nil {
+							return "", false, err
+						}
+						actual := resp.GetBackup().GetStatus().Status
+						if actual == requiredStatus {
+							return "", false, nil
+						}
+						return "", true, fmt.Errorf("backup status for [%s] expected was [%v] but got [%s]", backupName, requiredStatus, actual)
+					}
+					_, err = DoRetryWithTimeoutWithGinkgoRecover(backupCloudBackupMissingCheckFunc, 20*time.Minute, 30*time.Second)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verfiying backup %s is in CloudBackup missing state", backupName))
+				}(backupName)
+			}
+			wg.Wait()
+		})
+
+		Step("Verify if the restores for the above backups get failed", func() {
+			log.InfoD("Verify if the restores for the above backups get failed")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, backupName := range backupNames {
+				restoreName := fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+				appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, appNamespaces)
+				err = CreateRestoreWithValidation(ctx, restoreName, backupName, make(map[string]string), make(map[string]string), destinationClusterName, orgID, appContextsToBackup)
+				dash.VerifyFatal(strings.Contains(err.Error(), "CloudBackup objects are missing"), true, fmt.Sprintf("Verifying if the restore [%s] is getting Failed after JSON file deletion.", restoreName))
```

Comment:
how about we also inspect the restore and verify the status?

Reply:
Addressed the restore status inspection.

Reply:
@kphalgun-px I addressed this change, but I failed to realise that in our case the restore never happens, it's not that the restore fails, its that the restore request itself doesn't go through because the backup status is cloudfilemissing. I may have to revert the change.

> log message: Failed to inspect the restore. Err: restore with name 'tp-restore-single-ns-postgres-backup-pxb-0-86098-01-13-21h42m44s-jkomb-backup' not found for org 'default'

Reply:
Oh okay. Thanks!